pkg/kubectl/namespace_test.go

```diff
@@ -44,18 +45,48 @@ func TestNamespaceGenerate(t *testing.T) {
 			params:    map[string]interface{}{},
 			expectErr: true,
 		},
+		{
+			params: map[string]interface{}{
+				"name": 1,
+			},
+			expectErr: true,
+		},
+		{
+			params: map[string]interface{}{
+				"name": nil,
+			},
+			expectErr: true,
+		},
+		{
+			params: map[string]interface{}{
+				"name_wrong_key": "some_value",
+			},
+			expectErr: true,
+		},
+		{
+			params: map[string]interface{}{
+				"NAME": "some_value",
+			},
+			expectErr: true,
+		},
 	}
 	generator := NamespaceGeneratorV1{}
-	for _, test := range tests {
+	for index, test := range tests {
 		obj, err := generator.Generate(test.params)
-		if !test.expectErr && err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
-		if test.expectErr && err != nil {
-			continue
-		}
-		if !reflect.DeepEqual(obj.(*api.Namespace), test.expected) {
+		switch {
+		case test.expectErr && err != nil:
+			continue // loop, since there's no output to check
+		case test.expectErr && err == nil:
+			t.Errorf("%v: expected error and didn't get one", index)
+			continue // loop, no expected output object
+		case !test.expectErr && err != nil:
+			t.Errorf("%v: expected error and didn't get one", index)
+			continue // loop, no output object
+		case !test.expectErr && err == nil:
+		// do nothing and drop through
```

Comment:
go doesn't auto-fallthrough, you have specify it.  In this use, I think its a little more clear to end the `switch` here and keep the `if` around the deep equals separate.

Sorry, I didn't make this far down in my previous pass.